pkg/pillar/hypervisor/xen.go

```diff
@@ -497,21 +497,21 @@ func (ctx xenContext) Info(domainName string, domainID int) (int, DomState, erro
 	lastState := domainState[len(domainState)-1:]
 	log.Debugf("Info: domain: %s lastState: %s", domainName, lastState)
 	//if domainState is not 'r' or 'b' then the domain is not healthy.
-	stateMap := map[string]DomState{
-		"r": Running,
-		"b": Blocked,
-		"p": Paused,
-		"s": Exiting,
-		"c": Crashed,
-		"d": Dying,
+	stateMap := map[string]types.SwState{
+		"r": types.RUNNING,
+		"b": types.RUNNING,
+		"p": types.HALTED,
```

Comment:
Note that when we xl create the domU in halted state, so that we can do the "tune" before releasing it.
I wonder whether mapping "p" to HALTED risk reporting something confusing if it looks during that "tune" state.

Reply:
You're probably right @eriknordmark - I guess we do need a PAUSED state after all

Reply:
When xen is transitioning between states, sometimes it reports empty state (i.e. '------') which will result in `effectiveDomainState = types.UNKNOWN`.
This will cause a problem in domainmgr.go/verifyStatus(), to be more specific at the below section of that method:
```
		// check if any supporting tasks (like qemu) have crashed
		if configActivate && status.Activated && !(domainStatus == types.BOOTING ||
			domainStatus == types.RUNNING ||
			domainStatus == types.HALTING) {
			errStr := fmt.Sprintf("verifyStatus(%s) task has crashed (%d)", status.Key(), domainStatus)
			log.Errorf(errStr)
			status.SetErrorNow("one of the application's tasks have crashed - please restart application instance")
			status.Activated = false
			status.State = types.HALTED
			publishDomainStatus(ctx, status)
			if err := hyper.Task(status).Delete(status.DomainName, status.DomainId); err != nil {
				log.Errorf("failed to delete domain: %s (%v)", status.DomainName, err)
			}
		}
```
Here when domainStatus = types.UNKNOWN, we will delete the domain.

Reply:
Funny you should mention this @adarsh-zededa -- this is EXACTLY the bug I fixed yesterday after running some tests. If only this comment was done a day earlier -- it would've saved me an hour or so ;-)